### PR TITLE
ghost locator maints loot

### PR DESCRIPTION
**Pull request by Ilya246**
## About the PR
adds an item that works like the anomaly locator, but for ghosts

reacts to non-admin ghosts and revenants in a 12-tile radius

is very rare maints loot, took me 110 maints lockers in testing to roll just one

has limited battery and needs periodic charging

## Why / Balance
i think this would be quite fun to have while not significantly affecting game balance, ghosts can lead you places with it, but they also can (and likely will) troll you; the rarity further minimises effect on game balance, so i think the fun is worth any potential problems this may (not that i think it will) cause

discord discussion (including some maintainers) at https://discord.com/channels/310555209753690112/1008709214006427689/1286339786369405018 viewed it favourably

as for the AI eye or revenant detecting aspect, i don't believe the item can reasonably be used to help people find/avoid either of those

if maintainers/comments discussion think that this shouldn't be obtainable, i will instead make it an admeme item and remove it from maints pool

## Technical details
adds a marker component to mark the entities it spots

## Media
tested, works

https://github.com/user-attachments/assets/6f3728bf-7a97-4416-a419-bdf1f26d7c9e

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: Spectral locator, rare maintenance loot that lets you know if any ghosts are nearby.

### CODE REVIEW

**Comment by lzk228** [comment]
as i can see it also works on aghosts, don't think it's quite good, requires a discussion

**Comment by Ilya246** [comment]
> as i can see it also works on aghosts, don't think it's quite good, requires a discussion

could move the marker component from aghosts to the specific entities this is supposed to detect then

**Comment by lzk228** [comment]
ask admins in discord first, what they think about it

**Comment by superjj18** [comment]
Put in a sci research tree, now chap can become an exorcist lmao

**Comment by baddiePRO** [comment]
> Put in a sci research tree, now chap can become an exorcist lmao

more ghost interaction would be so fun, give us a ouija board that you can talk to ghost gang with

**Comment by Nimfar11** [comment]
Perhaps also give traitors a device that can show that they are being followed by a specific AI?

**Comment by Ilya246** [comment]
> Put in a sci research tree, now chap can become an exorcist lmao

would have to be a tech with 1 item which is kinda eh, or could alternatively put it into the same tech as the anomaly locator but i don't think it quite fits

**Comment by superjj18** [comment]
> > Put in a sci research tree, now chap can become an exorcist lmao
> 
> 
> 
> would have to be a tech with 1 item which is kinda eh, or could alternatively put it into the same tech as the anomaly locator but i don't think it quite fits

Lmao I just think it'd be cool for the chap to have a way to access this sort of functionality for giggles.  Doesn't need to be a part of this pr, but maybe the item can be added to chaplain's loadout item choices or give the Bible a chaplain-only function similar

**Comment by Everturning** [comment]
i think the chaplain vs ghosts thing should be put in some sort of chaplain gameplay PR.

**Comment by Ilya246** [comment]
sprite added (nobody said it's bad when i asked people so it seems it's okay) and description updated to fit the sprite, should be good for review

File: Content.Shared/Ghost/SpectralComponent.cs
Comment (slarticodefast) [inline Content.Shared/Ghost/SpectralComponent.cs:9]:
```suggestion
public sealed partial class SpectralComponent : Component { }
```

File: Content.Shared/Ghost/SpectralComponent.cs
Comment (slarticodefast) [inline Content.Shared/Ghost/SpectralComponent.cs:3]:
```suggestion
namespace Content.Shared.Ghost;
```
import not needed

**Review by slarticodefast — CHANGES_REQUESTED** [review]

**Comment by slarticodefast** [comment]
Hey, we had a discussion and vote among the maintainers and think this addition is good with the following adjustments:
- Remove SpectralComponent from the AI and aghosts
- Make it a little harder for the ghosts to answer to yes/no questions by moving closer to the device and making it beep louder. Currently it is easy to transmit messages with that method, causing potential metagaming problems due to the ghost vision and dead chat. Maybe increase the maximum radius a little and decrease the maximum beeping rate, if that is enough.

**Comment by Ilya246** [comment]
> Hey, we had a discussion and vote among the maintainers and think this addition is good with the following adjustments:
> 
>     * Remove SpectralComponent from the AI and aghosts
> 
>     * Make it a little harder for the ghosts to answer to yes/no questions by moving closer to the device and making it beep louder. Currently it is easy to transmit messages with that method, causing potential metagaming problems due to the ghost vision and dead chat. Maybe increase the maximum radius a little and decrease the maximum beeping rate, if that is enough.

done, all tested, only detects revenants and normal ghosts

min beeping interval 0.1s -> 0.25s, range 8 -> 12 so very easy to troll by other ghosts

**Comment by Plykiya** [comment]
Maybe someday I'll be able to enjoy Phasmophobia in SS14 if enough ghost related tools get added...

**Comment by Ilya246** [comment]
> Hey, we had a discussion and vote among the maintainers and think this addition is good with the following adjustments:
> 
>     * Remove SpectralComponent from the AI and aghosts
> 
>     * Make it a little harder for the ghosts to answer to yes/no questions by moving closer to the device and making it beep louder. Currently it is easy to transmit messages with that method, causing potential metagaming problems due to the ghost vision and dead chat. Maybe increase the maximum radius a little and decrease the maximum beeping rate, if that is enough.

my opinion on this is that the communication thing might be okay since this is very rare anyway and it's likely only up to person per round is going to have it; i've already PRed this to goobstation separately, *and the chaplain gets one roundstart there,* to see how it turns out, and i don't see ghosts using it to communicate even with the 8 radius and fast max beeping, plus the AI and aghost detection doesn't see usage since you can't distinguish whether it's a ghost or something else

**Comment by Everturning** [comment]
plus its fun and not meta to use a ghost detector to detect ghosts

**Comment by ScarKy0** [comment]
i mean there could definitely be a syndie item that does this same thing except just for the AI

**Comment by slarticodefast** [comment]
I gave the new values a test and it seems good to me.
The sprite still needs Art Director approval, but otherwise this should be good to merge.

**Review by slarticodefast — APPROVED** [review]

File: Content.Shared/Ghost/SpectralComponent.cs
Comment (metalgearsloth) [inline Content.Shared/Ghost/SpectralComponent.cs:7]: Not networked

File: Resources/Prototypes/Entities/Objects/Fun/spectral_locator.yml
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Fun/spectral_locator.yml:29]: This is going to query every single entity in a 12 tile range, I don't think proximity detector code is suitable for this yet without causing performance issues.

**Review by metalgearsloth — DISMISSED** [review]

**Comment by metalgearsloth** [comment]
See https://github.com/space-wizards/space-station-14/issues/26790

File: Resources/Prototypes/Entities/Objects/Fun/spectral_locator.yml
Comment (slarticodefast) [reply Resources/Prototypes/Entities/Objects/Fun/spectral_locator.yml:29]: Doesn't the anomaly locator do the same on a much wider radius?

Comment (Ilya246) [reply Resources/Prototypes/Entities/Objects/Fun/spectral_locator.yml:29]: yeah, and this is much rarer than anomaly locators so should probably be fine

**Comment by Ilya246** [comment]
git decided to revolt for a moment

**Comment by slarticodefast** [comment]
I further discussed it with Sloth and it should be fine since it is much rarer and has a smaller radius than the anomaly locator.
The ProximityDetectorComponent still really needs a refactor though.
The sprite got art director approval, so this is good to merge again.
Sorry for the delay and thank you for your contribution!

**Review by slarticodefast — APPROVED** [review]

**Review by chromiumboy — APPROVED** [review]
Code looks fine and tested it in-game

I did find a bug in proximity detectors unrelated to this PR (#32857) but that shouldn't be cause to hold this up further

**Comment by baddiePRO** [comment]
let's fucking gooo ghost locator merged

**Pull request merged**
